doc/en/array.txt

```diff
@@ -505,24 +554,41 @@
       AEval( <aArray>, <bBlock>, [<nStart>], [<nCount>] ) --> aArray
    $ARGUMENTS$
       <aArray> Is the array to be evaluated.
+
       <bBlock> Is a code block to evaluate for each element processed.
+
       <nStart> The beginning array element index to evaluate.
+
       <nCount> The number of elements to process.
    $RETURNS$
       <aArray> an array pointer reference.
    $DESCRIPTION$
-      This function will evaluate and process the subscript elements
-      in <aArray>. A code block passed as <bBlock> defines the operation
-      to be executed on each element of the array. All elements in <aArray>
-      will be evaluated unless specified by a beginning subscript position
-      in <nStart> for <nCount> elements.
-      Two parameters are passed to the code block <bBlock>. The individual
-      elements in an array are the first parameter and the subscript position
-      is the second.
-      AEval() does not replace a FOR...NEXT loop for processing arrays. If
-      an array is an autonomous unit, AEval() is appropriate. If the array
-      is to be altered or if elements are to be reevaluated, a FOR...NEXT
-      loop is more appropriate.
+      This function evaluates the elements of <aArray>, and executes for each
+      of them the processing that's defined with <bBlock>. 
+      By default, all the elements of array are being processed, starting from
+      1st and up to the last element. 
+      
+      If  <nStart> and/or <nCount> parameters are given, then the processing 
+      starts from <nStart> element and continues for the next <nCount> elements 
+      (if defined) or up to the last element of the array.
+
+      The <bBlock> code block receives two parameters: the element value and 
+      element's index (position) into the array. i.e. {|xValue, nIndex| ... }
+      Worth to note that elements are passed to code block 'by value', 
+      thus any change being made to this value doesn't  affects the value of 
+      element in the array.
+   $EXAMPLES$   
+      PROCEDURE Main
+         LOCAL a := { 10, 20, 30 }
+         AEval( a, {|e, n| Qout( e * n+1 , a[ n ] )} )
+         Qout( a[1], a[2], a[3] ) // array elements unchanged
+         ? "----"
+         AEval( a, {|e, n| Qout( e * n+1, a[n] *= n+1 )}, 2, 1 )
+         // here the 2nd element been changed, because we've explicitly used 
+         // its pointer 'a[n] *=..' into array.
+         Qout( a[1], a[2], a[3] ) 
+         WAIT
+         RETURN
```

Comment:
Function name casing can be fixed using a `hbmk2 -fixcase array.txt` command. Spaces around operators, and inside `[]` and `{}` brackets needs to be done manually (or using `hbformat`, though this can sometimes do more harm than good).